src/renderer/components/CanvasVisualiser/Events.js

```diff
@@ -114,25 +139,37 @@ export function onClick(params) {
 export function onContext(params) {
   const nodeId = this.getNetwork().getNodeAt(params.pointer.DOM);
   if (nodeId && !selectedNodes) {
-    this.showLabelsOnEdgesConnectedToNode(nodeId);
+    this.bringFocusToEdgesConnectedToNode(nodeId);
     this.getNetwork().selectNodes([nodeId], true);
   }
 }
 
 export function onHoverNode(params) {
   const nodeId = params.node;
-  const connectedEdges = this.getNetwork().getConnectedEdges(nodeId);
+  const connectedEdgeIds = this.getNetwork().getConnectedEdges(nodeId);
+  const connectedEdges = this.getEdges(connectedEdgeIds);
   let connectedNodes = this.getNetwork().getConnectedNodes(nodeId).concat(nodeId);
-  // Show labels on connected edges
-  this.showEdgeLabels(connectedEdges);
+
+  if (connectedEdges.length) {
+    const shouldAlwaysShowLabel = connectedEdges[0].options.label && connectedEdges[0].options.label.show;
+    if (!shouldAlwaysShowLabel) {
+      // this.showEdgeLabels(connectedEdges);
+    }
+
+    const shouldAlwaysShowArrow = connectedEdges[0].options.arrow && connectedEdges[0].options.arrow.show;
+    if (!shouldAlwaysShowArrow) {
+      // this.showEdgeArrows(connectedEdges);
```

Comment:
do we still need this if?

Reply:
shoot! yes, we do. uncommenting it ...

Reply:
review your own pr before asking for a review